### DISCUSSION TITLE
.write_videofile problems

### DISCUSSION THREAD

**Discussion by mryt66** [Zulko/moviepy#2089] [Q&A]
I did install ffmpeg with conda console and newest magick.exe idk what's going wrong, can't figure it out
final_clip = CompositeVideoClip(videos)
final_clip = final_clip.set_fps(60.00)
final_clip.write_videofile("oussst.mp4")
final_clip.close()

---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[19], line 42
     40 final_clip = CompositeVideoClip(videos)
     41 final_clip = final_clip.set_fps(60.00)
---> 42 final_clip.write_videofile("oussst.mp4")
     43 final_clip.close()

File D:\Anaconda\Lib\site-packages\decorator.py:232, in decorate.<locals>.fun(*args, **kw)
    230 if not kwsyntax:
    231     args, kw = fix(args, kw, sig)
--> 232 return caller(func, *(extras + args), **kw)

File D:\Anaconda\Lib\site-packages\moviepy\decorators.py:54, in requires_duration(f, clip, *a, **k)
     52     raise ValueError("Attribute 'duration' not set")
     53 else:
---> 54     return f(clip, *a, **k)

File D:\Anaconda\Lib\site-packages\decorator.py:232, in decorate.<locals>.fun(*args, **kw)
    230 if not kwsyntax:
    231     args, kw = fix(args, kw, sig)
--> 232 return caller(func, *(extras + args), **kw)

File D:\Anaconda\Lib\site-packages\moviepy\decorators.py:135, in use_clip_fps_by_default(f, clip, *a, **k)
    130 new_a = [fun(arg) if (name=='fps') else arg
    131          for (arg, name) in zip(a, names)]
    132 new_kw = {k: fun(v) if k=='fps' else v
    133          for (k,v) in k.items()}
--> 135 return f(clip, *new_a, **new_kw)

File D:\Anaconda\Lib\site-packages\decorator.py:232, in decorate.<locals>.fun(*args, **kw)
    230 if not kwsyntax:
    231     args, kw = fix(args, kw, sig)
--> 232 return caller(func, *(extras + args), **kw)

File D:\Anaconda\Lib\site-packages\moviepy\decorators.py:22, in convert_masks_to_RGB(f, clip, *a, **k)
     20 if clip.ismask:
     21     clip = clip.to_RGB()
---> 22 return f(clip, *a, **k)

File D:\Anaconda\Lib\site-packages\moviepy\video\VideoClip.py:300, in VideoClip.write_videofile(self, filename, fps, codec, bitrate, audio, audio_fps, preset, audio_nbytes, audio_codec, audio_bitrate, audio_bufsize, temp_audiofile, rewrite_audio, remove_temp, write_logfile, verbose, threads, ffmpeg_params, logger)
    292 if make_audio:
    293     self.audio.write_audiofile(audiofile, audio_fps,
    294                                audio_nbytes, audio_bufsize,
    295                                audio_codec, bitrate=audio_bitrate,
    296                                write_logfile=write_logfile,
    297                                verbose=verbose,
    298                                logger=logger)
--> 300 ffmpeg_write_video(self, filename, fps, codec,
    301                    bitrate=bitrate,
    302                    preset=preset,
    303                    write_logfile=write_logfile,
    304                    audiofile=audiofile,
    305                    verbose=verbose, threads=threads,
    306                    ffmpeg_params=ffmpeg_params,
    307                    logger=logger)
    309 if remove_temp and make_audio:
    310     if os.path.exists(audiofile):

File D:\Anaconda\Lib\site-packages\moviepy\video\io\ffmpeg_writer.py:213, in ffmpeg_write_video(clip, filename, fps, codec, bitrate, preset, withmask, write_logfile, audiofile, verbose, threads, ffmpeg_params, logger)
    211     logfile = None
    212 logger(message='Moviepy - Writing video %s\n' % filename)
--> 213 with FFMPEG_VideoWriter(filename, clip.size, fps, codec = codec,
    214                             preset=preset, bitrate=bitrate, logfile=logfile,
    215                             audiofile=audiofile, threads=threads,
    216                             ffmpeg_params=ffmpeg_params) as writer:
    218     nframes = int(clip.duration*fps)
    220     for t,frame in clip.iter_frames(logger=logger, with_times=True,
    221                                     fps=fps, dtype="uint8"):

File D:\Anaconda\Lib\site-packages\moviepy\video\io\ffmpeg_writer.py:88, in FFMPEG_VideoWriter.__init__(self, filename, size, fps, codec, audiofile, preset, bitrate, withmask, logfile, threads, ffmpeg_params)
     77 self.ext = self.filename.split(".")[-1]
     79 # order is important
     80 cmd = [
     81     get_setting("FFMPEG_BINARY"),
     82     '-y',
     83     '-loglevel', 'error' if logfile == sp.PIPE else 'info',
     84     '-f', 'rawvideo',
     85     '-vcodec', 'rawvideo',
     86     '-s', '%dx%d' % (size[0], size[1]),
     87     '-pix_fmt', 'rgba' if withmask else 'rgb24',
---> 88     '-r', '%.02f' % fps,
     89     '-an', '-i', '-'
     90 ]
     91 if audiofile is not None:
     92     cmd.extend([
     93         '-i', audiofile,
     94         '-acodec', 'copy'
     95     ])

TypeError: must be real number, not NoneType

**Comment by SohamTilekar** [comment]
Could You Provide the Python Version, Moviepy Version, Platform and more.

**Reply by mryt66** [reply]
3.11.5 - Python
Anaconda, Inc. | Jupyter notebook
'1.0.3'- Moviepy version

import moviepy
from moviepy.editor import *
from moviepy.video.io.VideoFileClip import VideoFileClip

Also I discovered that before new ffmpeg installed the creation of simple video worked well, now even that isn't working, so I reinstalled ffmpeg as before but it didn't help neither

**Comment by SohamTilekar** [comment]
In the python 3.11 in moviepy almost all set_ methods do not work Full Bug Report And Way Around on the #2084 .

**Reply by mryt66** [reply]
I deleted all set_ methods and it is still this same problem

**Reply by SohamTilekar** [reply]
I Didn't Understand What You Say.

**Reply by mryt66** [reply]
from moviepy.editor import *

clip = VideoFileClip("Template.mp4").subclip(0, 5)
videos = [clip]
clip.write_videofile("ss.mp4")

even this code with all needed libraries gives this same error 
I guess the problem is in ffmpeg but on ffmpeg forum they send me here so I am a little bit stuck

**Reply by SohamTilekar** [reply]
No Idea It Work on my laptop.

Input Video: -

https://github.com/Zulko/moviepy/assets/152298897/aaad45cf-e3b3-4275-a51e-dc5eae082e21

Output Video: -

https://github.com/Zulko/moviepy/assets/152298897/70e96831-df26-4f0b-a7da-3f747920ab0d



Python 3.12 (.venv : venv)
windows 10
